code/game/objects/items/blueprints.dm

```diff
@@ -4,239 +4,74 @@
 	icon = 'icons/obj/items/blueprints.dmi'
 	icon_state = "blueprints"
 	attack_verb = list("attacked", "bapped", "hit")
-	var/const/AREA_ERRNONE = 0
-	var/const/AREA_STATION = 1
-	var/const/AREA_SPACE =   2
-	var/const/AREA_SPECIAL = 3
 
-	var/const/BORDER_ERROR = 0
-	var/const/BORDER_NONE = 1
-	var/const/BORDER_BETWEEN =   2
-	var/const/BORDER_2NDTILE = 3
-	var/const/BORDER_SPACE = 4
-
-	var/const/ROOM_ERR_LOLWAT = 0
-	var/const/ROOM_ERR_SPACE = -1
-	var/const/ROOM_ERR_TOOLARGE = -2
+	var/valid_z_levels = list()
+	var/area_prefix
 
 /obj/item/blueprints/Initialize()
 	. = ..()
 	desc = "Blueprints of the [station_name()]. There is a \"Classified\" stamp and several coffee stains on it."
+	
+	if(set_valid_z_levels())
+		set_extension(src, /datum/extension/eye/blueprints)
 
-/obj/item/blueprints/attack_self(mob/M)
-	if (!istype(M,/mob/living/carbon/human))
-		to_chat(M, "This stack of blue paper means nothing to you.")//monkeys cannot into projecting
-
+/obj/item/blueprints/attack_self(var/mob/living/carbon/human/user)
+	if (!istype(user))
```

Comment:
`if(!istype(user) || !user.check_dexterity(DEXTERITY_KEYBOARDS)` or something would be closer to monkeycheck.

Reply:
Might also be worth adding SPAN_WARNING while you're in the region.